### ISSUE TITLE
can't save classifierDL in spark-nlp 2.7 on Databricks runtime 6.5ML 

### ISSUE THREAD

**Issue by dlopes78** [JohnSnowLabs/spark-nlp#2061]
## Description

Saving pipeline for classifierDL was working in spark-nlp 2.6.5 but not in spark-nlp 2.7.0

## Expected Behavior

regression issue, was working in 2.6.5

## Current Behavior

org.tensorflow.TensorFlowException: not an sstable (bad magic number)

## Steps to Reproduce

val documentAssembler = new DocumentAssembler()
   .setInputCol("text")
   .setOutputCol("document")

val token = new Tokenizer()
  .setInputCols("document")
  .setOutputCol("token")

val embeddings = WordEmbeddingsModel.pretrained()
    .setInputCols("document", "token")
    .setOutputCol("embeddings")

val sentence_embeddings = new SentenceEmbeddings()
   .setInputCols("document", "embeddings")
   .setOutputCol("sentence_embeddings")
   .setPoolingStrategy("AVERAGE")

val docClassifier = new ClassifierDLApproach()
  .setInputCols("sentence_embeddings")
  .setOutputCol("class")
  .setLabelColumn("label")
  .setMaxEpochs(30)
  .setLr(5e-3f)
  .setDropout(0.5f)

val pipeline = new Pipeline()
  .setStages(
    Array(
      documentAssembler,
      token,
      embeddings,
      sentence_embeddings,
      docClassifier
    )
  )

val pipelineModel = pipeline.fit(trainDataset)
pipelineModel.write.overwrite().save("....file_path...")

## Context

2.6.5 had an existing issue where classifierDL model saved and the loaded back was giving issues, https://github.com/JohnSnowLabs/spark-nlp/issues/973. Was testing if it got fixed in 2.7.0 but now can't save the model in the first place.  

## Your Environment
Databricks runtime 6.5ML

**Comment by maziyarpanahi** [comment]
Hi,

could you please paste the full error when you are trying to save? Also, what is the exact path? (dbfs or FileStore)

**Comment by maziyarpanahi** [comment]
I can confirm this issue. The release fixed the issue of training classifiers in the clusters and immediately using them without any drop in their accuracy, but introduced an issue with saving them on Databricks local storage file.

I'll add this to be fixed in the next hot-fix release.

**Comment by maziyarpanahi** [comment]
@dlopes78 I have a Fat JAR which can be installed easily on Databricks (drag and drop) to test a fix for this issue. Please let me know if you are interested in participating. 

- The PyPI package will be the same
- Uninstall the Maven package of Spark NLP
- restart the cluster to be sure it is removed
- upload the JAR via the UI in libraries
- restart again
- and just test it

The tests passed on both Databricks AWS and Azure, but it's hard to pin point it down. It would be great if you can test it on your cluster as well. (the 6.5 was deprecated so I had to test on 6.4)

**Comment by dlopes78** [comment]
@maziyarpanahi I'm happy to test, let me know how I can help.

**Comment by dlopes78** [comment]
@maziyarpanahi  I also noticed that I can't run inference on the model just trained (so without saving it) - while it was working in 2.6.5:

val testDataset = spark.createDataFrame(Seq(
  (0, "some text")
)).toDF("id", "text")

val prediction = pipelineModel.transform(testDataset)

display(prediction)

SparkException: Failed to execute user defined function($anonfun$dfAnnotate$1: (array<array<struct<annotatorType:string,begin:int,end:int,result:string,metadata:map<string,string>,embeddings:array<float>>>>) => array<struct<annotatorType:string,begin:int,end:int,result:string,metadata:map<string,string>,embeddings:array<float>>>)
.....

Caused by: org.tensorflow.TensorFlowException: not an sstable (bad magic number)
	 [[{{node save/RestoreV2}}]]
	at org.tensorflow.Session.run(Native Method)

**Comment by maziyarpanahi** [comment]
Yes, that is the issue in 2.7.0:

Here is the Fat JAR, you can follow those steps and making sure the Fat JAR is being used and not the Maven package:

https://s3.amazonaws.com/auxdata.johnsnowlabs.com/public/spark-nlp-tf-fix20.jar

**Comment by dlopes78** [comment]
@maziyarpanahi I tested the Jar and the issue seems to be solved:
--train a model
-- test model and accuracy is correct
-- save and reload the model
--  use reloaded model

Thanks for the quick reply!

**Comment by maziyarpanahi** [comment]
That's great, since the JAR you have is actually a 2.7.0 release with this fix please go ahead and use it, please. I would like to be sure it doesn't come back since Databricks behave differently on local storage from time to time.

**Comment by maziyarpanahi** [comment]
This should be resolved in the 2.7.1 release https://github.com/JohnSnowLabs/spark-nlp/releases/tag/2.7.1
Please re-open if the issue persists.
